site/docs/references/custom-selectors.md

```diff
@@ -0,0 +1,181 @@
+---
+id: references/custom-selectors
+title: Custom Selectors
+layout: docs
+---
+
+You use [Custom selectors](https://drafts.csswg.org/css-extensions/#custom-selectors) to define an alias that can match complex selectors. 
+
+For example, a specific type of button within a form that appears only when hovered can be defined as a custom selector. By defining the button as a custom selector with its own name, the button is exposed in the API and available for use by using just the custom selector name.
+
+You could also use custom selectors to define a group of selectors with one name. For example, you can access all the headings on a page as one custom selector. This could be useful if you want to style just their color the same.
+
+## Usage
+
+The following code maps the alias name `controlBtn` that matches any `.btn` CSS class nested under the `.controls` CSS class.
+
+****
+```css
+/* CSS */
+@namespace "Comp";
+@custom-selector :--controlBtn .controls .btn;
+/* 
+selector: .Comp__root .Comp__controls .Comp__btn 
+*/
+:--controlBtn { border: 1px solid grey; }
+/* 
+selector: .Comp__root .Comp__controls .Comp__btn:hover 
+*/
+:--controlBtn:hover { border-color: red; }
+```
+
+### Expose pseudo-element
+
+Custom selectors generate a [pseudo-element](./pseudo-elements.md). So, for example, [importing](./imports.md) a stylesheet into another stylesheet enables access to the `form-input` pseudo-element.
+
+```css
+/* CSS */
+@namespace "Page";
+:import {
+    -st-from: "./comp.st.css";
+    -st-default: Comp;
+}
+/*
+selector: .Page__root .Comp__root .Comp__controls .Comp__btn
+*/
+Comp::controlBtn { 
+    background: gold; 
+}
+```
+
+> **Note**:    
+> If a `custom-selector` alias conflicts with a local CSS class name, the exposed `pseudo-element` targets the `custom-selector`. However, the exported CSS class is still exported to JavaScript.
+
+## Use cases
+
+The following examples demonstrate how to effectively use custom selectors in **Stylable**.
+
+### Container and recursive components
+
+Some components might contain nested instances of themselves because they're a container or a "recursive" component. For example, a tree component might render itself. 
+
+If the component exposes any `pseudo-elements`, it is a good practice to define them as `custom selectors` with [child selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_selectors) to avoid effecting the inner parts of internal instances.
+
+The following example shows how a tree component exposes an icon.
+
+```css
+/* CSS */
+@namespace "Tree";
+@custom-selector :--icon .root > .icon;
+```
+
+Here you can use the icon `custom selector` from the outside just like you would use a CSS class selector or `pseudo-element`.
```

Comment:
from the outside just like you would use any other `pseudo-element`.

Reply:
Done